Update syntax to be compatible with Rails 4

jheiss opened the pull request:
This clears up all the deprecation warnings under Rails 4.  It is not backwards compatible with Rails 3.  I don't believe there is syntax for these operations that works in both.  So for backwards compatibility we'd have to wrap everything if "if rails >= 4 else end" blocks and that seemed horrible.  Given the otherwise slow pace of changes it seems reasonable to just tell Rails 3 folks to stay on the last released gem.

jheiss committed:
Update syntax to be compatible with Rails 4

laserlemon commented on test/dag_test.rb:
```diff
@@ -1,8 +1,8 @@
+gem 'activerecord', '~> 4.0.2'
 require "./init"
-
+I18n.enforce_available_locales = true
```
This probably doesn't belong in acts-as-dag.

jheiss replied:
I just put it in the test file to silence some warnings.  Now that rails 4.0.4 is out it would no longer be needed.

laserlemon replied:
Could we rip this out? Would love for the rest of this to be merged. :smiley:

jheiss replied:
OK, I was wrong.  Setting this is no longer needed in a full blown Rails app >= 4.0.4, but you still get the warning here where we're just using activerecord.  I'm fine with removing it, but given that it's just in the test suite it seems harmless.

resgraph replied:
Agreed. Harmless is fine.

On 3/24/14, 2:25 PM, Jason Heiss wrote:

> In test/dag_test.rb:
> 
> > ##  require "./init"
> > 
> > +I18n.enforce_available_locales = true
> 
> OK, I was wrong. Setting this is no longer needed in a full blown 
> Rails app >= 4.0.4, but you still get the warning here where we're 
> just using activerecord. I'm fine with removing it, but given that 
> it's just in the test suite it seems harmless.
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/resgraph/acts-as-dag/pull/13/files#r10907286.

## 

Signature
Robert A. Schmitt
Chief Executive Officer

_Concept *_Yard *
bright ideas, bright minds.
*
*
+1 (805) 403-0832  /  Cellular

laserlemon replied:
:ok_hand:

laserlemon commented on lib/dag/dag.rb:
```diff
@@ -93,12 +93,15 @@ def count_was
 
-    scope :ancestor_nodes, :joins => :ancestor
-    scope :descendant_nodes, :joins => :descendant
+    def ancestor_nodes
```
You're defining instance methods here when the previous scope was providing class methods.

laserlemon replied:
I would handle these like you handled the previous scopes:

``` ruby
scope :ancestor_nodes, lambda { joins(:ancestor) }
scope :descendant_nodes, lambda { joins(:descendant) }
```

jheiss replied:
Quite right.  I must have made some mistake in the syntax when I was working on this that made me think that scope wouldn't work, but what you suggest seems to work fine.  I'll update the pull request.

laserlemon commented:
:+1: to this change. It would require a major version bump because the new association conditions syntax in backwards incompatible. The gemspec should also scope the `activerecord` runtime dependency to `"~> 4.0"` and the development `activerecord` dependency can be dropped.

jheiss committed:
Convert ancestor and descendant_nodes to scope

Something made me think these wouldn't work as scopes, but @laserlemon
pointed out on the pull request that they would.

jheiss commented on lib/dag/dag.rb:
```diff
@@ -93,12 +93,11 @@ def count_was
       include Standard
     end
 
-    # TODO: rename? breaks when using 'where' query because :direct scope name and :direct => true parameter conflict?
```
It's possible I misunderstood the relevance of this comment and should not have removed it

laserlemon commented:
:+1: :shipit: :exclamation:

laserlemon commented:
@resgraph Is there anything more you'd like to see in order to have this merged and v4 released? Thank you!

resgraph commented:
No, I will merge it, test it and bump the version.

Thanks.

On 3/28/14, 10:10 PM, Steve Richert wrote:

> @resgraph https://github.com/resgraph Is there anything more you'd 
> like to see in order to have this merged and v4 released? Thank you!
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/resgraph/acts-as-dag/pull/13#issuecomment-38987224.

## 

Signature
Robert A. Schmitt
Chief Executive Officer

_Concept *_Yard *
bright ideas, bright minds.
*
*
+1 (805) 403-0832  /  Cellular

laserlemon commented:
Thank you! :clap:

laserlemon commented:
Ping. :grin:

zenkalia commented:
:ship: :shipit: :ship:

:wink:

resgraph commented:
Thanks for the reminders. Will carve out time today. Promise.

On 4/8/14, 11:51 AM, Mike Kendall wrote:

> :ship::shipit::ship:
> 
> :wink:
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/resgraph/acts-as-dag/pull/13#issuecomment-39886804.

## 

Signature
Robert A. Schmitt
Chief Executive Officer

_Concept *_Yard *
bright ideas, bright minds.
*
*
+1 (805) 403-0832  /  Cellular

resgraph commented:
acts-as-dag is pushed to RubyGems.org. Sorry it took so long. Dealing with Heartbleed the last two days.

zenkalia commented:
:bow: :birthday: :bow:
